plugins/woocommerce/src/Internal/Admin/Orders/MetaBoxes/CustomMetaBox.php

```diff
@@ -92,16 +92,51 @@ private function render_custom_meta_form( array $metadata_to_list, \WC_Order $or
 	 * Compute keys to display in autofill when adding new meta key entry in custom meta box.
 	 * Currently, returns empty keys, will be implemented after caching is merged.
 	 *
-	 * @param array|null         $keys Keys to display in autofill.
 	 * @param \WP_Post|\WC_Order $order Order object.
 	 *
 	 * @return array|mixed Array of keys to display in autofill.
 	 */
-	public function order_meta_keys_autofill( $keys, $order ) {
-		if ( is_a( $order, \WC_Order::class ) ) {
+	public function order_meta_keys_autofill( $order ) {
+		if ( ! is_a( $order, \WC_Order::class ) ) {
 			return array();
 		}
 
+		/**
+		 * Filters values for the meta key dropdown in the Custom Fields meta box.
+		 *
+		 * Compatibility filter for `postmeta_form_keys` filter.
+		 *
+		 * @since 6.9.0
+		 *
+		 * @param array|null $keys Pre-defined meta keys to be used in place of a postmeta query. Default null.
+		 * @param \WC_Order  $order The current post object.
+		 */
+		$keys = apply_filters( 'postmeta_form_keys', null, $order );
+
+		$limit = 0;
```

Comment:
Is this supposed to be attached to the `postmeta_form_limit` filter hook? Right now looks like `if ( $limit )` conditional a few lines down will always return false.

Reply:
Yeah. I made a mess when initially committing the code. It's been addressed in 8e843a0682d2428eab20909392b9b8a1ba48182f.